### DISCUSSION TITLE
autorun cell

### DISCUSSION THREAD

**Discussion by JoseBlanca** [jupyterlite/jupyterlite#1361] [Q&A] [answered]
Hi,

In my laboratory we are thinking about implementing a web app to do population genetics practices. The idea of being able to implement something like that in python and distributing it as a static web site is very attractive. It would allow us to use the code that we already have implemented to do the population genetic simulations and to allow teachers around the world to run the practices without installing any software and in any device.

We have implemented a first draft. You can take a look at it here:

https://bioinf.comav.upv.es/pop_lab/lab/index.html

To create the UI we're using pywidget. However, we're using Jupyter in an unconventional way. Eventually, it would be ideal to have the text with the practice explanation written using markdown cells and only one Python cell, the one required to run the application and start the UI. The student wouldn't need to now that the underlying functionality is written in Python, ideally it would be just a web application.

Thus, it would be helpful, in this case, to be able to auto-run the Python cell that creates the UI.

We understand that auto-running Jupyter cells is not ideal, because of security reasons. However, in this case the code is running in the browser's webassembly environment, so it might be acceptable, or maybe not, I don't know enough about this tech. Could you give me any advice or suggestion?

Best,

Jose Blanca

**Comment by jtpio** [comment, marked as answer]
Thanks @JoseBlanca for sharing, this looks neat!

> ideally it would be just a web application

Maybe you could have a look at Voici: https://github.com/voila-dashboards/voici

Voici is like [Voila](https://github.com/voila-dashboards/voila) but for JupyterLite. Meaning you can create a static web application with Python running the browser. 

To have a better idea or for inspiration you can also have a look at the gallery: https://voila-dashboards.github.io/voici-gallery/
